ci-operator/step-registry/network/mtu-migration/network-mtu-migration-commands.sh

```diff
@@ -0,0 +1,363 @@
+  done
+}
+
+print_debug() {
+  cat << HEADER
+
+################################
+  ____  _____ ____  _   _  ____ 
+ |  _ \| ____| __ )| | | |/ ___|
+ | | | |  _| |  _ \| | | | |  _ 
+ | |_| | |___| |_) | |_| | |_| |
+ |____/|_____|____/ \___/ \____|
+                                
+################################
+
+HEADER
+
+  trap - EXIT
+  oc get co
+  oc get mcp
+  oc get nodes
+  print_all_host_data
+  trap "print_debug_on_error" EXIT
+
+  cat << FOOTER
+
+####################################################
+  _____ _   _ ____    ____  _____ ____  _   _  ____ 
+ | ____| \ | |  _ \  |  _ \| ____| __ )| | | |/ ___|
+ |  _| |  \| | | | | | | | |  _| |  _ \| | | | |  _ 
+ | |___| |\  | |_| | | |_| | |___| |_) | |_| | |_| |
+ |_____|_| \_|____/  |____/|_____|____/ \___/ \____|
+
+####################################################
+
+FOOTER
+}
+
+print_debug_on_error() {
+  e=$?
+  [ $e -ne 0 ] && print_debug
+  exit $e
+}
+
+log "Applying MTU offset ${MTU_OFFSET} to the cluster"
+
+trap "print_debug_on_error" EXIT
+
+setup_packet_cluster
+
+wait_for_co "600s"
+
+print_debug
+
+log "Getting the current cluster network MTU..."
+cluster_mtu=$(oc get network.config --output=jsonpath='{.items..status.clusterNetworkMTU}')
+if [ -z "${cluster_mtu}" ]; then
+  log "Unable to get clusterNetworkMTU"
+  exit 1
+fi
+log "Cluster network MTU is ${cluster_mtu}"
+
+# Get host default gateway interface & MTU...
+node=$(oc get nodes -o jsonpath='{.items[0].metadata.name}')
+get_node_host_data "${node}"
+
+if [[ ${MTU_OFFSET} -ne 0 ]]; then
+  from=${cluster_mtu}
+  to=$((from+MTU_OFFSET))
+  host_to=$((host_mtu+MTU_OFFSET))
+  oc patch Network.operator.openshift.io cluster --type='merge'   --patch '{"spec":{"migration":null}}'
+  wait_for_cno_cleanup "60s"
+  log "Starting MTU migration from cluster network MTU ${from} and host MTU ${host_mtu} to cluster network MTU ${to} and host MTU ${host_to}"
+  oc patch Network.operator.openshift.io cluster --type='merge' --patch "{\"spec\": { \"migration\": { \"mtu\": { \"network\": { \"from\": ${from}, \"to\": ${to} } , \"machine\": { \"to\" : ${host_to}} } } } }"
+else
+  network_type=$(oc get network.config --output=jsonpath='{.items..status.networkType}')
+  if [ -z "${network_type}" ]; then
+    log "Unable to get networkType"
+    exit 1
+  fi
+  network_config="ovnKubernetesConfig"
+  if [ "${network_type}" = "OpenShiftSDN" ]; then
+    network_config="openshiftSDNConfig"
+  fi
+
+  host_to=$(oc get network.config --output=jsonpath='{.items..status.migration.mtu.machine.to}')
+  if [ -z "${host_to}" ]; then
+    log "Unable to get migration.host.to"
+    exit 1
+  fi
+
+  log "Ending MTU migration to cluster network MTU ${cluster_mtu}"
+  oc patch MachineConfigPool master --type='merge' --patch '{ "spec": { "paused": true } }'
+  oc patch MachineConfigPool worker --type='merge' --patch '{ "spec":{ "paused" :true } }'
+  # we have observed that if we don't space out these updates, MCO will miss one
```

Comment:
nothing to poll on instead?

Reply:
nothing that we know of for this specific thing. But then we do poll on `wait_for_final_mc`. The consequence though if we don't space out these updates, `wait_for_final_mc` waits 10 minutes or more. If we do space them it takes a lot of less time. But is not a job-will-fail consequence.

Reply:
ok, makes sense.

Reply:
It's dues this bug https://bugzilla.redhat.com/show_bug.cgi?id=2005694

Reply:
good find. maybe add this bug in the comment so someone might notice it later and be able to remove
the added delays when the bug is fixed.